src/iop/watermark.c

```diff
@@ -311,8 +370,71 @@ static gchar *_watermark_get_svgdoc(dt_iop_module_t *self, dt_iop_watermark_data
       svgdata = svgdoc;
     }
 
-    // Current image ID
+    // Simple text from watermark module
     gchar buffer[1024];
+
+    if (p->font)
+    {
+      g_snprintf(buffer, sizeof(buffer), "%s", p->text);
+      svgdoc = _string_substitute(svgdata, "$(WATERMARK_TEXT)", buffer);
+      if(svgdoc != svgdata)
+      {
+        g_free(svgdata);
+        svgdata = svgdoc;
+      }
+
+      g_snprintf(buffer, sizeof(buffer), "%02x%02x%02x",
+                 (unsigned)(p->color[0]*255), (unsigned)(p->color[1]*255), (unsigned)(p->color[2]*255));
```

Comment:
Will use clamp() as gdk_rgba_to_string() returns a format "rgb(r,g,b)" not supported by svg.